content/docs/aws/authentication-oidc.md

```diff
@@ -0,0 +1,129 @@
+
+<img src="/docs/images/aws/github-oauth-app-register-details.png"
+  alt="Github Oauth App Register Details"
+  class="mt-3 mb-3 border border-info rounded">
+
+After completing the form click **Register application** to proceed.
+
+### 3. Get your GitHub app's Client ID and Client Secret
+Once the application is registered, your app's `Client ID` and `Client Secret` will be displayed on the following page:
+
+<img src="/docs/images/aws/github-oauth-app-client-id-secret.png"
+  alt="Github Oauth App Client ID Secret"
+  class="mt-3 mb-3 border border-info rounded">
+  
+### 4. Copy your GitHub app's Client ID and Client Secret
+
+Go to your [Auth0 Dashboard](https://manage.auth0.com/dashboard/), if you haven't sign in with Auth0 yet, you can login at first.
+
+<img src="/docs/images/aws/auth0-login.png"
+  alt="Auth0 Login"
+  class="mt-3 mb-3 border border-info rounded">
+ 
+Then you can look at below page and modify the domain name Under `TENANT DOMAIN` to be the domain name you set up in Step 2.
+
+<img src="/docs/images/aws/auth0-welcome-page.png"
+  alt="Auth0 Welcome Page"
+  class="mt-3 mb-3 border border-info rounded">
+ 
+After modification, click **NEXT** button. Then you can set up your personal information there and click **CREATE ACCOUNT** button.
+ 
+After log into Auth0 Dashboard, select **Connections > Social**, then choose Github. Copy the `Client ID` and `Client Secret` from the `Developer Applications` of your app on Github into the fields on this page on Auth0.
+
+<img src="/docs/images/aws/auth0-github-setup.png"
+  alt="Auth0 Welcome Page"
+  class="mt-3 mb-3 border border-info rounded">
+
+Then click **SAVE** button.
+
+### 5. Set up Auth0 Callback URLs
+If you have already installed kubeflow by following [this tutorial](https://www.kubeflow.org/docs/aws/deploy/install-kubeflow/) before, run below command and you can copy ALB Hostname:
+```
+kubectl get ingress istio-ingress -n istio-system
+NAME            HOSTS   ADDRESS                                                                 PORTS   AGE
+istio-ingress   *       322ac077-istiosystem-istio-2af2-786120677.us-west-2.elb.amazonaws.com   80      56s
+```
+
+On Auth0 dashboard, click `Applications` on the left navigation bar, modify your ALB Hostname with prefix `https://` and postfix `/oauth2/idpresponse` in `Allowed Callback URLs` as below image:
+  
+<img src="/docs/images/aws/auth0-callback-url.png"
+  alt="Auth0 App Callback URL"
+  class="mt-3 mb-3 border border-info rounded">
+
+## Set up ALB with Auth0
+
+The general tutorial is [here](https://medium.com/@sandrinodm/securing-your-applications-with-aws-alb-built-in-authentication-and-auth0-310ad84c8595) you can follow.
+
+The details of instructions are listed below:
+
+### ALB Setup
```

Comment:
Hi @PatrickXYS,

I think you misunderstand what I want you to help. We want user to use kfctl to setup OIDC, similarly like what we did for cognito, that's why I give you instructions here 

```
oidc:
    oidcIssuer: https://kubeflow.auth0.com/
    oidcAuthorizationEndpoint: https://kubeflow.auth0.com/authorize
    oidcTokenEndpoint: https://kubeflow.auth0.com/oauth/token
    oidcUserInfoEndpoint: https://kubeflow.auth0.com/userinfo

    # this should be ACM's certArn rather than cognito user pool arn
    certArn: arn:aws:acm:us-west-2:xxxx:certificate/c7a9eb05-0975-4fc8-bb79xxxxx97e4031c
    oAuthClientId: yours
    oAuthClientSecret: yours
```

The reason I ask you to do it manually is because kfctl is not ready at that time.

Reply:
Yes, it's my misunderstanding. And it's the confusion that I want to sync with you on Friday, to make it more clear. I'm working on it right now.